File: Solutions/0169- Majority Element/Solution.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        my_dict = dict()
        n = len(nums) // 2

        for num in nums:
            if num not in my_dict:
                my_dict[num] = 1
            else:
                my_dict[num] += 1

            if my_dict[num] > n:
                return num

        return -1

File: Solutions/0169- Majority Element/Solution.java
import java.util.Arrays;

class Solution {
    public int majorityElement(int[] nums) {
        Arrays.sort(nums);
        int left = 0;
        int n = nums.length;

        while (left < nums.length) {
            int counter = 0;
            int number = nums[left];

            while (left < nums.length && nums[left] == number) {
                left++;
                counter++;
            }

            if (counter > (n/2))
                return number;
        }

        return -1;
    }
}